pr_script.sh

```diff
@@ -1,29 +1,94 @@
+#!/bin/bash
 
+###
+# EXIT CODES
+# 0 - success
+# 1 - parameter error
+###
 
-# git specifics
-REPO_URL=https://github.com/bird-house/finch
-PROJECT_NAME=finch
-COMMIT_MESSAGE="bump_finch_0.5.3"
-BRANCH_NAME="bump_finch_0.5.3"
+# Sample environment variables
+# REPO_URL=https://github.com/bird-house/birdhouse-deploy
+# PROJECT_NAME=birdhouse-deploy
+# DOCKERHUB_REPO="pavics/weaver"
+# IMAGE_ID="weaver-worker"
+# BUMP_TAG="WEAVER_WORKER_IMAGE"
+# BUMP_TAG_VALUE="1.13.2-worker"
+# BUMP_FILE="birdhouse/default.env"
+# GITHUB_USER=""
+# GITHUB_PASSWORD=""
 
-# bumpversion config
-IMAGE_VERSION_LOCATOR="FINCH_IMAGE"
-NEW_IMAGE_TAG="0.5.3"
-VERSION_FILE="birdhouse/default.env"
+REQUIRED_ENV_VARS='
+    REPO_URL
+    PROJECT_NAME
+    DOCKERHUB_REPO
+    IMAGE_ID
+    BUMP_TAG
+    BUMP_TAG_VALUE
+    BUMP_FILE
+    GITHUB_USER
+    GITHUB_PASSWORD
+'
 
-# hub config
-GITHUB_USER=""
-GITHUB_PASSWORD=""
+# args parsing
+for env_var in $REQUIRED_ENV_VARS
+do
+    if [[ ! -v "${env_var}" ]]; then
+        echo "[ERROR] [$0] Missing ${env_var} environment variable. Exiting."
+        exit 1
+    fi
+done
 
+WORKING_DIR="working_dir"
+
+echo "[STEP] [$0] [$PROJECT_NAME] Fetch tags"
+
+COMMIT_MESSAGE="bump ${IMAGE_ID} to ${BUMP_TAG_VALUE}"
+BRANCH_NAME="bump_${IMAGE_ID}_to_${BUMP_TAG_VALUE}"
+
+
+# prepare repo
+mkdir -p $WORKING_DIR
+# rm -rf $WORKING_DIR/$PROJECT_NAME
+cd $WORKING_DIR
+
+if [[ ! -d "$PROJECT_NAME" ]]
+then
+    echo "[INFO] Not existing project directory, cloning"
+    git clone $REPO_URL
+    cd $PROJECT_NAME
+else
+    echo "[INFO] Existing project directory, cleaning"
+    cd $PROJECT_NAME
+    git clean -fd
+    git reset --hard origin
+    git checkout master
+    git pull
+    git branch -D $BRANCH_NAME     # TODO : currently avoids already existing branch error. Handle error instead
+fi
 
-git clone $REPO_URL
-cd $PROJECT_NAME
 git checkout -b $BRANCH_NAME
 
-IMAGE_VERSION_LOCATOR=$IMAGE_VERSION_LOCATOR NEW_IMAGE_TAG=$NEW_IMAGE_TAG VERSION_FILE=$VERSION_FILE ./bump_version.sh
+# for testing purpose only, since birdhouse-deploy's default.env doesn't has WEAVER_WORKER_IMAGE variable for now
+if [[ ! -z "${TEST_ENV_CONFIG}" ]] && [[ "$BUMP_TAG" == "WEAVER_WORKER_IMAGE" ]]; then
+    echo "export WEAVER_WORKER_IMAGE=\"pavics/weaver:1.13.1-worker\"" >> $BUMP_FILE
+fi
 
+# bumpversion
+DOCKERHUB_REPO=$DOCKERHUB_REPO BUMP_TAG=$BUMP_TAG BUMP_TAG_VALUE=$BUMP_TAG_VALUE BUMP_FILE=$BUMP_FILE ../../bump_version.sh
+
+# to track last diff result
+git diff &> ../../last-diff-result.log
+
+# commit
 git add -A
-git commit -m $COMMIT_MESSAGE
-git push
+git commit -m "$COMMIT_MESSAGE"
 
-hub pull-request -m $COMMIT_MESSAGE
\ No newline at end of file
+# when in test mode, we can avoid to make a real PR
+if [[ ! -z "${EXIT_BEFORE_PR}" ]]; then
+    echo "[INFO] Exiting before PR, since EXIT_BEFORE_PR is defined. Exiting."
+    exit 0
+fi
+
+# PR
+git push
```

Comment:
git push using which credentials?

Reply:
Now using the GITHUB_TOKEN, the `hub` cli handle the auth nicely. Tested with real PR and no auth issues.